specification/app/resource-manager/Microsoft.App/preview/2023-11-02-preview/CommonDefinitions.json

```diff
@@ -1176,6 +1180,20 @@
         }
       }
     },
+    "KeyVaultProperties": {
+      "description": "Key vault reference using managed identity.",
+      "type": "object",
+      "properties": {
+        "identity": {
+          "description": "Resource ID of a managed identity to authenticate with Azure Key Vault, or System to use a system-assigned identity.",
+          "type": "string"
```

Comment:
Please check [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details) if it is an ARM Resource Id.

Reply:
to keep consistency with our another secret design:  https://github.com/Azure/azure-rest-api-specs/blob/main/specification/app/resource-manager/Microsoft.App/preview/2023-08-01-preview/CommonDefinitions.json#L563C12-L572   we expect the identity should either be a user assigned managed identity id or special string "system" or empty string. We already have a lot of customer using this way. So it is not always an arm id. I will DM you about our usage case.